lib/VMwareWebService/MiqVimVm.rb

```diff
@@ -720,8 +720,15 @@ def resetCd
 	#    In this case, backingFile must be the path to the existing VMDK.
 	# If backingFile is just the datastore name, "[storage 1]" for example,
 	#    file names will be generated as appropriate.
+	# 
+	# thinProvisioned - if new disk is thin (grow on demand, allowing storage overcommitment) or thick (pre-allocated)
+	# dependent       - if new disk is dependent (usual one) or not (meaning no delta file and no live snapshots of running VM)
+	# persistent      - if new disk should really save changes or discard them on VM poweroff
+	# 
+	# P.S. Good overview of use cases for dependent / independent and persistent / nonpersistent disks
+	# can be found at http://cormachogan.com/2013/04/16/what-are-dependent-independent-disks-persistent-and-non-persisent-modes/
 	#
-	def addDisk(backingFile, sizeInMB, label=nil, summary=nil, thinProvisioned=false)
+	def addDisk(backingFile, sizeInMB, label=nil, summary=nil, thinProvisioned=false, dependent=false, persistent=true)
```

Comment:
Got it. I can modify the code. However, to be honest, I'm not sure, if it is worth doing.

The point is that the only direct call to addDisk is in EmsVmware.vm_add_disk - https://github.com/ManageIQ/manageiq/pull/2111/files#diff-d793ddbf6b846f0172e517d6049d67cdR353 All the other calls go through VmOrTempalte.add_disk proxy - and that's really good, since it allows to script for "generic VM" rather than for "VMware VM". So, I see two reasonable options.

*Option 1*

We are working with addDisk only. In this case:
* EmsVmware.vm_add_disk has options={} as a parameter and encapsulates addDisk params
* EmsVmware.invoke_vim_ws always works with quite long lists of parameters, and I see no reason to make addDisk the only exception from this convention

*Option 2*

We are going to change generic VmOrTempalte.add_disk to accept option={} parameter. In this case the value is clear. However, this change has bigger impact than just MiqVimVm.addDisk. Is it really ok for me to do it? Will it be accepted and merged?